x-pack/plugins/infra/public/hooks/use_kibana_timefilter_time.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback } from 'react';
+import { useUpdateEffect, useMount } from 'react-use';
+import { useKibanaContextForPlugin } from './use_kibana';
+import { TimeRange } from '../../../../../src/plugins/data/public';
+
+export const useKibanaTimefilterTime = ({
+  from: fromDefault,
+  to: toDefault,
+}: TimeRange): [typeof getTime, typeof setTime] => {
+  const { services } = useKibanaContextForPlugin();
+
+  const getTime = useCallback(() => {
+    return services.data.query.timefilter.timefilter.getTime({ from: fromDefault, to: toDefault });
+  }, [services.data.query.timefilter.timefilter, fromDefault, toDefault]);
+
+  const setTime = useCallback(
+    (timeRange: TimeRange) => {
+      services.data.query.timefilter.timefilter.setTime(timeRange);
+    },
+    [services.data.query.timefilter.timefilter]
+  );
```

Comment:
Does this wrapper fulfill a meaningful role? Would there be any downsides to directly returning the `setTime` function from the hook instead?

Reply:
Good point, we can just return the function directly. I originally had some different parameters in here, but in this final version it's pointless.